src/lib/statistics/statistics_objects/abstract_histogram.hpp

```diff
@@ -0,0 +1,256 @@
+      const std::optional<AllTypeVariant>& variant_value2 = std::nullopt) const override;
+
+  std::shared_ptr<AbstractStatisticsObject> scaled(const Selectivity selectivity) const override;
+
+  /**
+   * Derive a Histogram from this histograms splitting its bins so that both the current bounds as well as the bounds
+   * specified in @param additional_bin_edges are present.
+   */
+  std::shared_ptr<AbstractHistogram<T>> split_at_bin_bounds(
+      const std::vector<std::pair<T, T>>& additional_bin_edges) const;
+
+  /**
+   * @return [{bin_minimum(BinID{0}), bin_maximum(BinID{0})}, {bin_minimum(BinID{1}), bin_maximum(BinID{1})}, ...]
+   */
+  std::vector<std::pair<T, T>> bin_bounds() const;
+
+  /**
+   * Returns the number of bins actually present in the histogram.
+   * This number can be smaller than the number of bins requested when creating a histogram.
+   * See implementations of this method in specific histograms for details.
+   */
+  virtual BinID bin_count() const = 0;
+
+  /**
+   * Returns the number of values represented in the histogram.
+   * This is equal to the number of rows in the segment during the generation of the bins for the histogram,
+   * without null values.
+   */
+  virtual HistogramCountType total_count() const = 0;
+
+  /**
+   * Returns the number of distinct values represented in the histogram.
+   * This is equal to the number of distinct values in the segment during creation.
+   */
+  virtual HistogramCountType total_distinct_count() const = 0;
+
+  /**
+   * Returns the smallest value in the bin.
+   */
+  virtual T bin_minimum(const BinID index) const = 0;
+
+  /**
+   * Returns the largest value in a bin.
+   */
+  virtual T bin_maximum(const BinID index) const = 0;
+
+  /**
+   * Returns the number of values in a bin.
+   */
+  virtual HistogramCountType bin_height(const BinID index) const = 0;
+
+  /**
+   * Returns the number of distinct values in a bin.
+   */
+  virtual HistogramCountType bin_distinct_count(const BinID index) const = 0;
+
+  /**
+   * Returns the width of a bin.
+   * This method is specialized for strings to return a numerical width.
+   */
+  virtual HistogramWidthType bin_width(const BinID index) const;
+
+  /**
+   * Helper function
+   * @return {bin_minimum(index), bin_maximum(index), bin_height(index), bin_distinct_count(index)}
+   */
+  HistogramBin<T> bin(const BinID index) const;
+
+  /**
+   * Returns the share of the value range of a bin that are smaller (or equals) than `value`.
+   * This method is specialized for strings.
+   * @{
+   */
+  float bin_ratio_less_than(const BinID bin_id, const T& value) const;
+  float bin_ratio_less_than_equals(const BinID bin_id, const T& value) const;
+  /** @} */
+
+ protected:
+  Cardinality _invert_estimate(const Cardinality& estimate) const;
+  CardinalityAndDistinctCountEstimate _invert_estimate(const CardinalityAndDistinctCountEstimate& estimate) const;
```

Comment:
why doc internal functions that are 4 lines long and are better explained by their implementation anyway?

Reply:
So that people who have a first look at a class get an idea of what it is able to do. Of course you could look into the implementation, but that requires a much deeper understanding.

Reply:
but the non-public functions do not do anything for "people", they are *internal*? But lest we argue more, I added a comment:

```c++

  /**
   * Helper for implementing, e.g., LessThan as an inversion of GreaterThanEquals. 
   * @return total_[distinct_]count() - estimate 
   */
  Cardinality _invert_estimate(const Cardinality& estimate) const;
  std::pair<Cardinality, DistinctCount> _invert_estimate(const std::pair<Cardinality, DistinctCount>& estimate) const;
```